packages/headless/src/controllers/insight/category-facet/headless-insight-category-facet.ts

```diff
@@ -0,0 +1,233 @@
+
+  /**
+   * Sets the number of values displayed in the facet to the originally configured value.
+   * */
+  showLessValues(): void;
+
+  /**
+   * Enables the facet. I.e., undoes the effects of `disable`.
+   */
+  enable(): void;
+
+  /**
+   * Disables the facet. I.e., prevents it from filtering results.
+   */
+  disable(): void;
+
+  /**
+   * Provides methods to search the facet's values.
+   */
+  facetSearch: CategoryFacetSearch;
+
+  /**
+   * The state of the `Facet` controller.
+   * */
+  state: InsightCategoryFacetState;
+}
+
+export interface InsightCategoryFacetState extends CoreCategoryFacetState {}
+
+/**
+ * Creates a `InsightCategoryFacet` controller instance.
+ *
+ * @param engine - The insight engine.
+ * @param props - The configurable `InsightCategoryFacet` properties.
+ * @returns A `InsightCategoryFacet` controller instance.
+ * */
+export function buildInsightCategoryFacet(
+  engine: InsightEngine,
+  props: InsightCategoryFacetProps
+): InsightCategoryFacet {
+  const coreController = buildCoreCategoryFacet(engine, props);
+  const {dispatch} = engine;
+  const getFacetId = () => coreController.state.facetId;
+  const createNoopCategoryFacetSearch = () => {
+    return {
+      updateText() {},
+      showMoreResults() {},
+      search() {},
+      clear() {},
+      updateCaptions() {},
+      select() {},
+      singleSelect() {},
+      get state() {
+        return {
+          values: [],
+          isLoading: false,
+          moreValuesAvailable: false,
+          query: '',
+        };
+      },
+    };
+  };
+
+  const facetSearch = createNoopCategoryFacetSearch();
+
+  const {state, ...restOfFacetSearch} = facetSearch;
+
+  if (!loadInsightCategoryFacetReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  return {
+    ...coreController,
+
+    facetSearch: restOfFacetSearch,
+
+    toggleSelect(selection: CategoryFacetValue) {
+      coreController.toggleSelect(selection);
+      const analyticsAction = getToggleSelectInsightAnalyticsAction(
+        getFacetId(),
+        selection
+      );
+      dispatch(executeSearch(analyticsAction));
+    },
+
+    deselectAll() {
+      coreController.deselectAll();
+      dispatch(executeSearch(logFacetClearAll(getFacetId())));
+    },
+
+    sortBy(criterion: CategoryFacetSortCriterion) {
+      coreController.sortBy(criterion);
+      dispatch(
+        executeSearch(logFacetUpdateSort({facetId: getFacetId(), criterion}))
+      );
+    },
+
+    isSortedBy(criterion: CategoryFacetSortCriterion) {
+      return coreController.isSortedBy(criterion);
+    },
+
+    showMoreValues() {
+      coreController.showMoreValues();
+      dispatch(fetchFacetValues(logFacetShowMore(getFacetId())));
+    },
+
+    showLessValues() {
+      coreController.showLessValues();
+      dispatch(fetchFacetValues(logFacetShowLess(getFacetId())));
+    },
+
+    enable() {
+      coreController.enable();
+    },
+
+    disable() {
+      coreController.disable();
+    },
```

Comment:
Since we're using `...coreController` on line 150, there is no need to redefine the `enable` and `disable` functions.